rts/Sim/Weapons/Weapon.cpp

```diff
@@ -1203,6 +1203,20 @@ float CWeapon::GetRange2D(float yDiff) const
 	return (root1 > 0.0f) ? math::sqrt(root1) : 0.0f;
 }
 
+void CWeapon::UpdateRange(float val)
+{
+	float previousVal = range;
+	range = val;
+	if (val >= owner->maxRange) {
```

Comment:
it can happen, and in that case the else if below would trigger forcing a recalculation of the max range iteratively